src/python/pants/backend/python/rules/prepare_chrooted_python_sources.py

```diff
@@ -3,47 +3,62 @@
 
 from dataclasses import dataclass
 
-from pants.backend.python.rules.inject_init import InjectedInitDigest
-from pants.engine.fs import Digest, DirectoriesToMerge
+from pants.engine.fs import (
+  EMPTY_DIRECTORY_DIGEST,
+  Digest,
+  DirectoriesToMerge,
+  FileContent,
+  InputFilesContent,
+  Snapshot,
+)
 from pants.engine.legacy.graph import HydratedTarget, HydratedTargets
 from pants.engine.rules import rule
 from pants.engine.selectors import Get, MultiGet
+from pants.python.pex_build_util import identify_missing_init_files
 from pants.rules.core.strip_source_roots import SourceRootStrippedSources
 
 
 @dataclass(frozen=True)
 class ChrootedPythonSources:
-  digest: Digest
+  snapshot: Snapshot
```

Comment:
I changed this because `Snapshot` is generally much more useful to rule authors than `Digest`.